api/pkg/controller/seed-controller-manager/addon/addon_controller.go

```diff
@@ -543,3 +573,48 @@ func (r *Reconciler) ensureRequiredResourceTypesExist(ctx context.Context, log *
 func deleteCommand(ctx context.Context, kubeconfigFilename, manifestFilename string) *exec.Cmd {
 	return exec.CommandContext(ctx, "kubectl", "--kubeconfig", kubeconfigFilename, "delete", "-f", manifestFilename, "--ignore-not-found")
 }
+
+func setAddonCodition(a *kubermaticv1.Addon, condType kubermaticv1.AddonConditionType, status corev1.ConditionStatus) {
+	idx, cond := getAddonCondition(a, condType)
+	if cond == nil {
+		cond = &kubermaticv1.AddonCondition{}
+		cond.Type = condType
+		cond.Status = status
+		cond.LastHeartbeatTime = metav1.Now()
+		cond.LastTransitionTime = metav1.Now()
+		a.Status.Conditions = append(a.Status.Conditions, *cond)
+		return
+	}
+	if cond.Status != status {
+		cond.LastTransitionTime = metav1.Now()
+		cond.Status = status
+	}
+	cond.LastHeartbeatTime = metav1.Now()
+	a.Status.Conditions[idx] = *cond
+}
+
+func getAddonCondition(a *kubermaticv1.Addon, condType kubermaticv1.AddonConditionType) (int, *kubermaticv1.AddonCondition) {
+	for i, c := range a.Status.Conditions {
+		if c.Type == condType {
+			return i, &c
+		}
+	}
+	return -1, nil
+}
+
+func addonResourcesCreated(addon *kubermaticv1.Addon) bool {
+	_, cond := getAddonCondition(addon, kubermaticv1.AddonResourcesCreated)
+	if cond != nil && cond.Status == corev1.ConditionTrue {
+		return true
+	}
+	return false
+}
+
+func hasEnsureResoucesLabel(addon *kubermaticv1.Addon) bool {
+	if len(addon.Labels) != 0 {
+		if ensure := addon.Labels[addonEnsureLabelKey]; ensure == "true" {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
It's completely legal to read from uninitialized map (in case when it's empty or nil).
```suggestion
func hasEnsureResoucesLabel(addon *kubermaticv1.Addon) bool {
	return addon.Labels[addonEnsureLabelKey] == "true"
}
```